server/main.py

```diff
@@ -31,50 +33,62 @@ class Payload(BaseModel):
     tvib: Optional[float] = None
     trot: Optional[float] = None
     path_length: float
-    simulate_slit: Optional[int] = None
+    simulate_slit: Optional[int]
     mode: Literal["absorbance", "transmittance_noslit", "radiance_noslit", "transmittance", "radiance"]
     database: Literal["hitran", "geisa"]
     use_simulate_slit: bool = False
 
 
+def calculate_spectrum(payload):
+    global spectrum
```

Comment:
I think these globals could cause issues with multiple people invoking the API at around the same time right? I think it may be better to have each API endpoint be stateless, even if it's less efficient. The calculate spectrum operation is fast on the server though so it shouldn't be so bad on the user experience.

Reply:
Isn't it possible to have per-session variables ? 

This reminds me of #153 where some post-processing operations are extremely fast if done without having to recompute a spectrum, but requires the latest spectrum to have been stored (per session).

Reply:
We could have some in-memory state that caches the spectrum object for a specific combination of input parameters to avoid having to perform re-calculation such as when the user applies a slit to a spectrum already calculated. We don't need the cache to be user-specific as users inputing the same parameters should get the same spectrum object.

That said, I don't think the level of implementation complexity is worth it if the computation is already reasonably fast. As a first step, I'd recommend making the API stateless to make it simpler to implement the feature, and if we find that we want to optimize it after that, we can add it.

Reply:
This sounds good ! 

Now that you're saying this, a similar feature is already implemented in the Radis Backend. The `SpectrumFactory`  (machinery behind `calc_spectrum`) can be connected to a `SpecDatabase`, which is really only a local folder, to automatically retrieve a Spectrum if it's already there; and store it if it's not. 

This is not implemented at the `calc_spectrum` level, but it could be done quite easily (already asked for a few years ago in https://github.com/radis/radis/issues/135  ). The api would be something like  : 

```python
s = calc_spectrum(..., local_folder="~/radis_spectra")  # automatically retrieve spectra if needed, computes and stores then if not  
``` 

Advantages : 
- similarities & tests are kept at the Radis backend level; we're sure Radis-App is not mistakently returning a different spectrum it thoughts was the same 
- no need to change much in Radis-app 
- we'd probably only have to add a feature to keep only the N latest spectra ; or clean the cache periodically.

Reply:
Looks good. Definitely would want to make sure we only store the latest N spectra to avoid a malicious user from filling up the server's disk.

Reply:
their is any configuration i have to do or just add` local_folder="~/radis_spectra"` in cal_spectrum ()